bindings/python/linkgrammar.py

```diff
@@ -404,19 +492,60 @@ def __init__(self, sent):
             self.num = 0
             clg.sentence_parse(sent._obj, sent.parse_options._obj)
 
+        def has_valid(self):
+            return 0 < self.sent.num_valid_linkages()
+
+        def has_linkages(self):
+            return 0 < self.sent.num_linkages_found()
+
+        def linkages(self, limit=1):
+            if self.has_valid():
+                return self
+            return self.sent.sentence_linkages(self.sent, limit=limit)
+
         def __iter__(self):
-            if (0 == clg.sentence_num_valid_linkages(self.sent._obj)):
+            if self.has_valid():
+                return self
+            else:
                 return iter(())
-            return self
 
         def next(self):
-            if self.num == clg.sentence_num_valid_linkages(self.sent._obj):
+            if self.num == self.sent.num_valid_linkages():
                 raise StopIteration()
             linkage = Linkage(self.num, self.sent, self.sent.parse_options._obj)
             self.num += 1
             return linkage
 
         __next__=next      # Account python3
 
+    class sentence_linkages(object):
```

Comment:
I did not write a comment. Sorry. 
This class is for enumerating linkages found, which are no valid. It is possible in the command utility link-parser, when the following variable is turned on
bad          Display of bad linkages                               0 (Off)

It is useful for debugging input.

Evgeny